### PR TITLE
Add deterministic test for Jastrow optimization

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
Add deterministic test for Jastrow coefficient on the optimization process.
 This test is to check whether optimization process works normally via direct comparison of Jastrow coefficient with the reference. It will provide the easy way to find an issue in the optimization process (Ex. #1888).     

 He atom example is used as the test, and only one J2 coefficient is used as the test for simple test.

 The reference is generated with QMCPACK 3.6.0 with Intel18 compiler.

 Hyeondeok Shin

### CODE REVIEW

**Comment by ye-luo** [comment]
Please remove qmc.xml cont.xml stat.h5 files.

**Comment by Hyeondeok-Shin** [comment]
@ye-luo Yes, they are not necessary. Deleted.

File: tests/molecules/He_ae/check_opt.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    output = get_jub_b(prefix_file)
+    reference = get_jub_b(ref_file)
+
+    if len(output) != len(reference):
+       exit_fail('Number of coefficient in test({0}) does not match with the reference({1})'.format(len(output),len(reference)))
+
+    success = True
+   
+    for i in range(len(output)):
+       quant_success = float(output[i]) == float(reference[i])
```
Comment (jtkrogel) [inline tests/molecules/He_ae/check_opt.py]: This floating point comparison should have a tolerance.  E.g. abs(float(output[i])-float(reference[i]))<1e-5.

Comment (Hyeondeok-Shin) [reply tests/molecules/He_ae/check_opt.py]: @jtkrogel Yes, it's right. I added tolerance (1e-6) for the test, and now script prints tolerance and deviation from the reference.

**Comment by prckent** [comment]
More comments to follow, but is this confirmed to pass with different compilers, recent versions etc.?

**Comment by Hyeondeok-Shin** [comment]
@prckent 
1. Test was passed with QMCPACK develop version (gcc 6.2 and intel 18 on Intel Xeon) & QMCPACK 3.7 (BG/Q).
2. I'll add more commit for smaller optimization steps and the VMC run

**Comment by Hyeondeok-Shin** [comment]
@prckent 

1. Now Optimization steps become smaller.
2. VMC run has added in the end, and deterministic test for the VMC energy has added to check whether optimized parameter is successfully stored.

Please let me know if additional changes are needed.

**Comment by prckent** [comment]
OK to test

**Pull request merged**